include/base/periodic_boundary_base.h

```diff
@@ -145,6 +155,14 @@ class PeriodicBoundaryBase
    * pointer is treated as equivalent to the identity matrix.
    */
   std::unique_ptr<DenseMatrix<Real>> _transformation_matrix;
+
+  /**
+   * Imposing a periodic boundary uses a PointLocator to match points on
+   * the two sides of the boundary. This set specifies the subdomains
+   * that the PointLocator searches over. If it is empty, then it is
+   * ignored and we search over all subdomains.
+   */
```

Comment:
Could we move this comment to the setter function (in an ideal world users would just read about the APIs and ignore private internals) and add an example (or a pointer to the new example app) describing its use?